Added Lua Scripting workflow file

LB-ArturZieba opened the pull request:
Signed-off-by: Artur Zięba <86952082+LB-ArturZieba@users.noreply.github.com>

**Change summary:**
* Added Lua Scripting workflow file.

michleza commented:
Notes as I went through the LKG:

- Let's keep list items as bullet points in the requests column unless they are steps or a number of things the user should be aware of. I think the idea of an LKG workflow is that 'Requests' are suggestions for things to test around as the user navigates that area of the project.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Create a Lua skeleton script | <ol><li>Create a new .lua file by selecting *File → New* in Lua Editor, make sure to save it in a path accessible to Editor (for example *<project path\>\\Assets\\*</li><li>Add a skeleton Lua script to the file, as seen here: https://www.o3de.org/docs/user-guide/scripting/lua/basic-lua-script/#skeleton-script - remember to replace *BaseScript* in the file to the n ...
```
Request 1: Missing closing parentheses .
Request 2: Remove the "as seen here: link" text and change to doc reference link over the "skeleton Lua script" `(skeleton Lua script)[https://www.o3de.org/docs/user-guide/scripting/lua/basic-lua-script/#skeleton-script]`

LB-ArturZieba replied:
1. Added parentheses.
2. Modified the webpage link as described.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
````diff
@@ -0,0 +1,293 @@
+| Create a Lua skeleton script | <ol><li>Create a new .lua file by selecting *File → New* in Lua Editor, make sure to save it in a path accessible to Editor (for example *<project path\>\\Assets\\*</li><li>Add a skeleton Lua script to the file, as seen here: https://www.o3de.org/docs/user-guide/scripting/lua/basic-lua-script/#skeleton-script - remember to replace *BaseScript* in the file to the n ...
+| Create an entity using the created skeleton Lua script | <ol><li>In Editor, create a new entity and add a *Lua Script* component to it</li><li>Select *Lua Script* component on your new entity, click the *Pick Lua Script* button and choose your skeleton script</li><li>Enter and exit Game Mode (using for example *Ctrl+G*) while paying attention to the Editor's Console</li></ol> | <ul><li>*Lua Scr ...
+
+**"Area: Creating a Lua skeleton script file" script:**
+```
+-- BaseScript.lua 
+
+local BaseScript = 
+{
+    Properties =
+    {
+        -- Property definitions
+    }
+}
+
+function BaseScript:OnActivate()
+     Debug.Log(tostring("Hi!"));
+end
+
+function BaseScript:OnDeactivate()
+     Debug.Log(tostring("Bye!"));
+end
+
+return BaseScript
+```
````
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

LB-ArturZieba replied:
Adjusted this and other similar code sections as described.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
````diff
@@ -0,0 +1,293 @@
+| Add a property to Properties table in a script | <ol><li>With Lua skeleton script opened (*Area: Lua skeleton script usage* script can be used for convenience), replace `-- Property definitions` with `My Value = 1,` </li><li>Add a line to OnActivate() that prints the property after a string (for example change `Debug.Log(tostring("Hi!"));` to `Debug.Log(tostring("Hi! My value is: " .. BaseScrip ...
+| Add local properties in a script | <ol><li>Add the following lines to the top of the OnDeactivate() function: <ul><li>`Bye = "Bye! "`</li><li>`NumberOne = 1`</li><li>`NumberTwo = 2`</li><li>`NumbersAdded = NumberOne + NumberTwo`</li></ul></li><li>Modify the Debug.Log line so that it uses the local properties - for example: `Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. N ...
+
+**"Area: Properties usage in Lua scripts" script:**
+```
+-- BaseScript.lua 
+
+local BaseScript = 
+{
+    Properties =
+    {
+        MyValue = 1,
+    }
+}
+
+function BaseScript:OnActivate()
+     Debug.Log(tostring("Hi! My value is: " .. BaseScript.Properties.MyValue));
+end
+
+function BaseScript:OnDeactivate()
+     Bye = "Bye! "
+     NumberOne = 1
+     NumberTwo = 2
+     NumbersAdded = NumberOne + NumberTwo
+     Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));
+end
+
+return BaseScript
+```
````
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+## Area: Debugging existing script
+
+### Project Requirements
+Any project with the Remote Tools Connection Gem added can be used and the Lua Editor has to be connected to Editor. Saved script from *Area: Properties usage in Lua scripts* helps with setup.
```
Blank line between section title & text body.

Let's turn this into a list. There's multiple things here to be done and each should be their own line.

LB-ArturZieba replied:
1. Added blank lines between this and other similar title/body text in the file.
2. Adjusted this and other Project Requirements to be lists.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Open Debugging tools | <ol><li>Open Breakpoints tool using any method (Toolbar Breakpoints button, or *View → Breakpoints*) </li><li>Add breakpoints to some lines on your script using any method (Select a line and click either Toggle Breakpoint button on toolbar, *Debug → Toggle Breakpoint* or press *F9*. Alternatively left-click the area next to the line number on the script) </li><li>Remove a ...
```
Request 1, 2, 4, & 5: Listify the methods

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Open Debugging tools | <ol><li>Open Breakpoints tool using any method (Toolbar Breakpoints button, or *View → Breakpoints*) </li><li>Add breakpoints to some lines on your script using any method (Select a line and click either Toggle Breakpoint button on toolbar, *Debug → Toggle Breakpoint* or press *F9*. Alternatively left-click the area next to the line number on the script) </li><li>Remove a ...
+| Basic breakpoint use | <ol><li>Add a breakpoint to a Debug.Log(); line in OnActivate() function (for example line 12 in the "Area: Properties usage in Lua scripts" script) </li><li>Make sure that Lua Editor is connected to the Editor and enter Game Mode</li><li>Change the window to the Lua Editor while the Game Mode is running (for example using Alt+Tab navigate to the Lua Editor window)</li><l ...
```
Request 1: Code wrap the code refences.
Request 4: Sub-Steps should listify the methods in which to perform actions.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+## Area: Properties usage in Lua scripts
+
+### Project Requirements
+Any project can be used. Saved script from *Area: Lua skeleton script usage* helps with setup.
```
Blank line between.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+## Area: Creating a Lua skeleton script file
+
+### Project Requirements
+Any project can be used.
```
Blank line between.

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+<b></b>
+## Area: Opening Lua Editor, connecting to Editor and using Find tool
+
+### Project Requirements
+Any project with the Remote Tools Connection Gem added can be used.
```
Blank line between.

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+- Windows
+- Linux
+
+### Docs:
+[O3DE Documentation: Writing Lua Scripts in Open 3D Engine](https://www.o3de.org/docs/user-guide/scripting/lua/) <br>
+[O3DE Documentation: Introduction to Open 3D Engine’s Lua Editor](https://www.o3de.org/docs/user-guide/scripting/lua/lua-editor/) <br>
+[O3DE Documentation: Basic Structure of a Lua Script](https://www.o3de.org/docs/user-guide/scripting/lua/basic-lua-script/) <br>
+[O3DE Documentation: Debugging Lua Scripts](https://www.o3de.org/docs/user-guide/scripting/lua/debugging-scripts/) <br>
+[O3DE Documentation: Debugging with Lua Editor](https://www.o3de.org/docs/user-guide/scripting/lua/debugging-tutorial/) <br>
+[O3DE Documentation: Using EBuses in Lua](https://www.o3de.org/docs/user-guide/scripting/lua/ebus/) <br>
```
Blank line between title & body.

Make these bullet points

LB-ArturZieba replied:
1. Addressed in earlier point.
2. Adjusted links to be in a bullet point list.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+2. Warnings or assets that appear in the Log
+3. Unresponsive UI elements 
+
+### Platforms:
+- Windows
```
Blank line between.

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+## Area: Moving entity using Lua script 
+
+### Project Requirements
+Any project with the Remote Tools Connection Gem added can be used and the Lua Editor has to be connected to Editor. Saved script from *Area: Properties usage in Lua scripts* helps with setup.
```
Blank line between title & body.

Listify and separate the actions going on here.

LB-ArturZieba replied:
1. Addressed in earlier point.
2. Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+|----------|----------|---------------------|
+| Create .inputbidings file | <ol><li>Open Asset Editor (for example *Tools → Asset Editor)* and create a new .inputbindings file (*File → New → Input Bindings*)</li><li>Add three following Input Event Groups to the .inputbindings file: <ul><li>Event Name: `Forward`; Event Generators: `keyboard_key_alphanumeric_W` with Event value multiplier: `1.0` and `keyboard_key_alphanumeric_S` with Event val ...
+| Modify the Lua script so that it uses keyboard input to move the entity | <ol><li>Modify the script as shown below the table in under the **"Area: Moving entity using Lua script" script:**</li><li>Find the following copied functions in the Class Reference (Remember that Lua Editor has to be connected to the Editor):<ul><li>GetEntityForward</li><li>AddVelocity</li><li>GetEntityRight</li><li>GetT ...
+| Create and move Player entity in Game Mode | <ol><li>Create an entity with the following components: *Mesh*, *Input*, *PhysX Character Controller*, *PhysX Character Gameplay*</li><li>Assign the PlayerMovement.inputbindings to the Input component</li><li>Assign the Lua script to the Lua Script component.</li><li>Add a Model Asset to the Mesh component (for example BeveledCylinder.fbx)</li><li>Ad ...
```
Request 1: Listify the components
Request 6: Wrap the coordinates in a code block `{X = 0.0; Y = -3.0; Z = 2.0}` to make it easy to spot.

LB-ArturZieba replied:
1. Addressed in earlier point.
2. Wrapped the coordinates in a code block.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
````diff
@@ -0,0 +1,293 @@
+--OnHeld is activated when an input is being held or applied constantly (as in the case of mouse movement)
+function PlayerScript:OnHeld(inputValue)
+	--Create a reference to the currently used input
+	local currentBusId = InputEventNotificationBus.GetCurrentBusId();
+	
+	--If "Forward" input is held (current used input is equal to "Forward") then
+	--Create a vector variable based on the entity's forward vector(EntityEntity_VM.GetEntityForward) with length equal to value of the input(inputValue)
+	--Add velocity (CharacterControllerRequestBus.Event.AddVelocity) equal to the above vector (self.forwardVector) to the entity housing this script (self.entityId)
+	if currentBusId == self.forwardBusId then
+		self.forwardVector = EntityEntity_VM.GetEntityForward(self.entityId, inputValue);
+		CharacterControllerRequestBus.Event.AddVelocity(self.entityId, self.forwardVector);
+	end
+	
+	--If "Strafe" input is held (current used input is equal to "Strafe") then
+	--Create a vector variable based on the entity's right vector(EntityEntity_VM.GetEntityRight) with length equal to value of the input(inputValue)
+	--Add velocity (CharacterControllerRequestBus.Event.AddVelocity) equal to the above vector (self.rightVector) to the entity housing this script (self.entityId)
+	if currentBusId == self.strafeBusId then
+		self.rightVector = EntityEntity_VM.GetEntityRight(self.entityId, inputValue);
+		CharacterControllerRequestBus.Event.AddVelocity(self.entityId, self.rightVector);
+	end
+	
+	--If "Look" input (horizontal mouse movement) is used (current used input is equal to "Forward") then
+	--Create a float variable that is a multiplication of the input value (inputValue) and time elapsed from the last tick (TickRequestBus.Broadcast.GetTickDeltaTime())
+	--Apply rotation around Z axis of entity housing this script (TransformBus.Event.RotateAroundLocalZ(self.entityId,...) equal to the above float (self.rotateSpeed)
+	if currentBusId == self.lookBusId then
+		self.rotateSpeed = inputValue * TickRequestBus.Broadcast.GetTickDeltaTime();
+		TransformBus.Event.RotateAroundLocalZ(self.entityId, self.rotateSpeed);
+	end
+end
+
+function BaseScript:OnDeactivate()
+     Bye = "Bye! "
+     NumberOne = 1
+     NumberTwo = 2
+     NumbersAdded = NumberOne + NumberTwo
+     Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));
+end
+
+return BaseScript
+```
````
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+## Area: Using ScriptEvents to communicate between two Lua scripts
+
+### Project Requirements
+Any project can be used.
```
Space between.

LB-ArturZieba replied:
Addressed in earlier point.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
````diff
@@ -0,0 +1,293 @@
+	self.ScriptEventHandler = SimpleScriptEvent.Connect(self)
+end
+
+--The delay setup is put in place so that the Send event is guaranteed to launch after all other entities/scripts (including the Receive event) are active
+
+--OnTick is activated each tick returning delta time (dt) since the last tick
+function SendEvent:OnTick(dt)
+	--Each tick subtract the time it took since the last tick from the current timer value and overwrite it (Can be looked at as "Count from 1 to 0 using ticks")
+	timer = timer - dt
+	--If timer variable reaches 0 or negative values then
+	if timer <= 0 then
+		--Send the SimpleScriptEvent's SendNumber event with a Number parameter 123
+		SimpleScriptEvent.Broadcast.SendNumber(123)
+		--Reset the timer variable to 1
+		timer = 1
+	end
+end
+
+return SendEvent
+```
+*ReceiveEvent.lua*
+```
+local ReceiveEvent = 
+{
+}
+
+function ReceiveEvent:OnActivate()
+	--Connect to the created Script Event (SimpleScriptEvent is the name of the Script Event file)
+	self.ScriptEventHandler = SimpleScriptEvent.Connect(self, self.entityId)
+
+end
+
+--Receive the SimpleScriptEvent's SendNumber event with it's parameter (number)
+function ReceiveEvent:SendNumber(number)
+	--Print the received parameter to the Console
+	Debug.Log("Received number: " .. tostring(number))
+end
+
+return ReceiveEvent
+```
````
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

LB-ArturZieba replied:
Addressed in earlier point.

LB-ArturZieba commented:
> Notes as I went through the LKG:
> 
> * Let's keep list items as bullet points in the requests column unless they are steps or a number of things the user should be aware of.
> * I think the idea of an LKG workflow is that 'Requests' are suggestions for things to test around as the user navigates that area of the project and not necessarily everything that needs to be completed for the workflow/timebox
> * Add some images for buttons since this editor's UI is pretty dense and it can be difficult to find some of the controls

I have addressed all of the above points:
1. This was addressed as a part of the changes requested by @FuzzyCarterAWS, inline text was changed to lists where appropriate.
2. Adjusted two "Requests" in the "Properties usage in Lua scripts" area as per this request.
3. Added images where the description could be ambiguous.

LB-ArturZieba commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,293 @@
+
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Launch Lua Editor and connect to Editor |<ol><li> Check different methods of entry to Lua Editor (*Tools → Lua Editor*, Lua Script component)</li> <li> Resize the Lua Editor window, move the present panes around and undock/dock them</li> <li> Connect to Editor by clicking *Target* button and choosing *Editor.exe (xyzxyzxy)* from the dropdown</li></ol>|<ul><li>Lua Editor launches and is stable < ...
```
1. This and all further similar entries were adjusted to be lists.
2. Reworded as described.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Docs:
+
+<ul>
+	<li>[O3DE Documentation: Writing Lua Scripts in Open 3D Engine](https://www.o3de.org/docs/user-guide/scripting/lua/)</li>
+	<li>[O3DE Documentation: Introduction to Open 3D Engine’s Lua Editor](https://www.o3de.org/docs/user-guide/scripting/lua/lua-editor/)</li>
+	<li>[O3DE Documentation: Basic Structure of a Lua Script](https://www.o3de.org/docs/user-guide/scripting/lua/basic-lua-script/)</li>
+	<li>[O3DE Documentation: Debugging Lua Scripts](https://www.o3de.org/docs/user-guide/scripting/lua/debugging-scripts/)</li>
+	<li>[O3DE Documentation: Debugging with Lua Editor](https://www.o3de.org/docs/user-guide/scripting/lua/debugging-tutorial/)</li>
+	<li>[O3DE Documentation: Using EBuses in Lua](https://www.o3de.org/docs/user-guide/scripting/lua/ebus/)</li>
+</ul>
```
Use MD formatting for lists outside of a table.

FuzzyCarterAWS replied:
As an aside.

HTML formatting is only required inside of MD tables. Otherwise MD list formatting works and is preferred.

LB-ArturZieba replied:
I have changed this and all other similar tables to use MD formatting instead of HTML as requested.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Project Requirements
+
+<ul><li>Any project with the Remote Tools Connection Gem</li></ul>
```
MD formatting

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Project Requirements
+
+<ul><li>Any project can be used</li><li>Saved script from "Area: Lua skeleton script usage" helps with setup</li></ul>
```
MD List

LB-ArturZieba replied:
Addressed in earlier request.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Project Requirements
+
+<ul><li>Any project</li></ul>
```
MD Formatting

LB-ArturZieba replied:
Addressed in earlier request.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Project Requirements
+
+<ul><li>Any project with the Remote Tools Connection Gem added</li><li>Lua Editor connected to Editor</li><li>Saved script from "Area: Properties usage in Lua scripts" helps with setup</li></ul>
```
MD list

LB-ArturZieba replied:
Addressed in earlier request.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Project Requirements
+
+<ul><li>Any project with the Remote Tools Connection Gem</li><li>Lua Editor connected to Editor</li><li>Saved script from "Area: Properties usage in Lua scripts" helps with setup</li></ul>
```
MD List

LB-ArturZieba replied:
Addressed in earlier request.

FuzzyCarterAWS commented on testing-guidance/workflow-tests/scripting/lua-scripting-tests.md:
```diff
@@ -0,0 +1,329 @@
+
+### Project Requirements
+
+<ul><li>Any project</li></ul>
```
MD List

LB-ArturZieba replied:
Addressed in earlier request.

michleza commented:
I like the revisions that have been made. This is criticism of github markdown and not your work but I wish the code snippets were easier to notice in their collapsed state. If you think you could draw more attention to them somehow that would be the next revision I would suggest.

michleza commented:
> 

I'm not familiar with other methods right now but I will do some quick research to see if anything appears. For now this should be fine though.

FuzzyCarterAWS commented:
It looks like the expand overrides the color formatting on the github rendering.

![image](https://user-images.githubusercontent.com/29718640/214174997-e630576b-5826-43d6-918f-355640fa174b.png)

A couple alternatives:
Rename the `"Area: Moving entity using Lua script" script:` pattern to something like `(Click to Expand) <filename>.lua`

Additionally, these files could be all grouped together at the top under Docs in a new section called "Script Files".  Or these could be put into an assets section for the workflow definition similar to what has been done in the [animation graph creation workflow](https://github.com/o3de/sig-simulation/blob/main/testing-guidance/workflow-tests/animation/animation-editor-workflows.md#area-animation-graph-creation) & [Script an Actor with Animation Workflow](https://github.com/o3de/sig-simulation/blob/main/testing-guidance/workflow-tests/animation/animation-lkg-workflows.md#area-script-an-actor-with-an-animation-graph-to-play-animations-via-user-input-during-runtime)

These solutions would require minor edits to the workflow text to be in-line with the changes, but would help with discoverability of the scripts, and you could even use the in-doc linking to link to the Script Files section for a user to quickly jump to where they need to be in the document to find the script file references.

LB-ArturZieba commented:
> @LB-ArturZieba, It looks like the expand overrides the color formatting on the github rendering.
> 
> ![image](https://user-images.githubusercontent.com/29718640/214174997-e630576b-5826-43d6-918f-355640fa174b.png)
> 
> A couple alternatives: Rename the `"Area: Moving entity using Lua script" script:` pattern to something like `(Click to Expand) <filename>.lua`
> 
> Additionally, these files could be all grouped together at the top under Docs in a new section called "Script Files". Or these could be put into an assets section for the workflow definition similar to what has been done in the [animation graph creation workflow](https://github.com/o3de/sig-simulation/blob/main/testing-guidance/workflow-tests/animation/animation-editor-workflows.md#area-animation-graph-creation) & [Script an Actor with Animation Workflow](https://github.com/o3de/sig-simulation/blob/main/testing-guidance/workflow-tests/animation/animation-lkg-workflows.md#area-script-an-actor-with-an-animation-graph-to-play-animations-via-user-input-during-runtime)
> 
> These solutions would require minor edits to the workflow text to be in-line with the changes, but would help with discoverability of the scripts, and you could even use the in-doc linking to link to the Script Files section for a user to quickly jump to where they need to be in the document to find the script file references.

Renamed the dropdowns as described and updated their relevant entries in the tables.

Added a section at the top of the file with links to all of the scripts, as well as a section with script link in each section that contains one.

FuzzyCarterAWS reviewed (approved):
LGTM!
